<html lang="en">
<head>
<title>Exception Handling - Debugging with GDB</title>
<meta http-equiv="Content-Type" content="text/html">
<meta name="description" content="Debugging with GDB">
<meta name="generator" content="makeinfo 4.13">
<link title="Top" rel="start" href="index.html#Top">
<link rel="up" href="Python-API.html#Python-API" title="Python API">
<link rel="prev" href="Basic-Python.html#Basic-Python" title="Basic Python">
<link rel="next" href="Values-From-Inferior.html#Values-From-Inferior" title="Values From Inferior">
<link href="http://www.gnu.org/software/texinfo/" rel="generator-home" title="Texinfo Homepage">
<meta http-equiv="Content-Style-Type" content="text/css">
<style type="text/css"><!--
  pre.display { font-family:inherit }
  pre.format  { font-family:inherit }
  pre.smalldisplay { font-family:inherit; font-size:smaller }
  pre.smallformat  { font-family:inherit; font-size:smaller }
  pre.smallexample { font-size:smaller }
  pre.smalllisp    { font-size:smaller }
  span.sc    { font-variant:small-caps }
  span.roman { font-family:serif; font-weight:normal; } 
  span.sansserif { font-family:sans-serif; font-weight:normal; } 
--></style>
<link rel="stylesheet" type="text/css" href="../cs.css">
</head>
<body>
<div class="node">
<a name="Exception-Handling"></a>
<p>
Next:&nbsp;<a rel="next" accesskey="n" href="Values-From-Inferior.html#Values-From-Inferior">Values From Inferior</a>,
Previous:&nbsp;<a rel="previous" accesskey="p" href="Basic-Python.html#Basic-Python">Basic Python</a>,
Up:&nbsp;<a rel="up" accesskey="u" href="Python-API.html#Python-API">Python API</a>
<hr>
</div>

<h5 class="subsubsection">22.2.2.2 Exception Handling</h5>

<p><a name="index-python-exceptions-1431"></a><a name="index-exceptions_002c-python-1432"></a>
When executing the <code>python</code> command, Python exceptions
uncaught within the Python code are translated to calls to
<span class="sc">gdb</span> error-reporting mechanism.  If the command that called
<code>python</code> does not handle the error, <span class="sc">gdb</span> will
terminate it and print an error message containing the Python
exception name, the associated value, and the Python call stack
backtrace at the point where the exception was raised.  Example:

<pre class="smallexample">     (gdb) python print foo
     Traceback (most recent call last):
       File "&lt;string&gt;", line 1, in &lt;module&gt;
     NameError: name 'foo' is not defined
</pre>
   <p><span class="sc">gdb</span> errors that happen in <span class="sc">gdb</span> commands invoked by Python
code are converted to Python <code>RuntimeError</code> exceptions.  User
interrupt (via <kbd>C-c</kbd> or by typing <kbd>q</kbd> at a pagination
prompt) is translated to a Python <code>KeyboardInterrupt</code>
exception.  If you catch these exceptions in your Python code, your
exception handler will see <code>RuntimeError</code> or
<code>KeyboardInterrupt</code> as the exception type, the <span class="sc">gdb</span> error
message as its value, and the Python call stack backtrace at the
Python statement closest to where the <span class="sc">gdb</span> error occured as the
traceback.

   </body></html>

